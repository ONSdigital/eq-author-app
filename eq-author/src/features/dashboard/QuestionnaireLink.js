import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { NavLink } from "react-router-dom";
import { buildQuestionnairePath } from "utils/UrlUtils";
import { colors, radius } from "constants/theme";
import CustomPropTypes from "custom-prop-types";

const StyledLink = styled(NavLink)`
  text-decoration: none;
  color: ${colors.blue};
  padding: 0.5em;
  &:focus {
    border-radius: ${radius};
    outline: none;
    box-shadow: 0 0 0 3px ${colors.tertiary};
  }
  &:hover {
    text-decoration: underline;
  }

  ${props =>
    props.disabled &&
    css`
      text-decoration: none;
      color: ${colors.text};
    `};
`;

const QuestionnaireLink = ({ questionnaire, disabled, ...otherProps }) => {
  return (
    <StyledLink
      {...otherProps}
      disabled={disabled}
      to={buildQuestionnairePath({ questionnaireId: questionnaire.id })}
    />
  );
};

QuestionnaireLink.defaultProps = {
  disabled: false
};

QuestionnaireLink.propTypes = {
  questionnaire: CustomPropTypes.questionnaire,
  disabled: PropTypes.bool
};

export default QuestionnaireLink;
