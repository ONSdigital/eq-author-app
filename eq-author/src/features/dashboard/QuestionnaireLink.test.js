import React from "react";
import { shallow } from "enzyme";
import QuestionnaireLink from "./QuestionnaireLink";

const questionnaire = {
  id: "1",
  title: "Foo",
  createdAt: "2017/01/02",
  sections: [
    {
      id: "1",
      pages: [{ id: "1" }]
    }
  ],
  createdBy: {
    name: "Alan"
  }
};

describe("QuestionnaireLink", () => {
  it("should render", () => {
    const wrapper = shallow(
      <QuestionnaireLink questionnaire={questionnaire} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
