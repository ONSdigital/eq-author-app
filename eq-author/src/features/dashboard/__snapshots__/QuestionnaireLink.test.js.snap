// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireLink should render 1`] = `
<QuestionnaireLink__StyledLink
  disabled={false}
  to="/questionnaire/1"
/>
`;
